src/utlist.h

```diff
@@ -600,6 +631,39 @@ do {
   }                                                                                            \
 } while (0)
 
+#define DL_INSERT_INORDER(head,add,cmp)                                                        \
+    DL_INSERT_INORDER2(head,add,cmp,prev,next)
+
+#define DL_INSERT_INORDER2(head,add,cmp,prev,next)                                             \
+do {                                                                                           \
+  LDECLTYPE(head) _tmp;                                                                        \
+  if (head) {                                                                                  \
+    DL_LOWER_BOUND(head,_tmp,add,cmp);                                                              \
+    DL_APPEND_ELEM(head,_tmp,add);                                                             \
+  } else {                                                                                     \
+    (head) = (add);                                                                            \
+    (head)->next = NULL;                                                                       \
+    (head)->prev = NULL;                                                                       \
```

Comment:
FYI: this should be `(head)->prev = (head)` to preserve utlist's invariant that `(head)->prev` in a doubly-linked list points to the tail of the list.